### DISCUSSION TITLE
How to check if a user is online?

### DISCUSSION THREAD

**Discussion by charlesoestergaard** [supabase/supabase#5378] [Questions] [answered]
Hello! I want to be able to check if a user is online or not.

This is so i can make a online indicator next to the profile image.

People login with email and password, not magic link.

What i mean
https://i.imgur.com/SWzlFRO.png

**Comment by GaryAustin1** [comment]
Just some thoughts, ideas...

First what is "online"?  Signed in?  Active?  Are you talking about the current user being connected to the database online, or seeing if a list of users is currently online?

If a single current user you can just realtime status to monitor connection to the database.

But I assume you want to see if other users are "online"...

Auth tables only have two indicators, first is last signed in time in auth.users, the other is the auth.refresh_tokens table for last token updated_at time.  You could trigger on update/inserts to refresh_tokens and write to a table in public with last refresh time for each user.  Then realtime monitor the status table for changes to refresh tokens or poll the table when needed with a recent updated_at time filter. 

If you want to know if users are connected or active you could have a "heartbeat" or update call in the client code to a status table with an updated_at time.  Then once again use realtime, or poll the table with a filter on updated_at for a recent time range.  Users who have not had the heartbeat or status update recently would not show up in filter.

Depending on number of users and how often you update/poll, this could get to be a real bandwidth/performance hog.

**Reply by charlesoestergaard** [reply]
Thanks for your answer, im new in supabase and this seems to be pretty complicated. Mabye someone allready done this, and know a good way.

**Comment by w3b6x9** [comment]
@rumfiske see https://github.com/supabase/supabase/discussions/5544#discussioncomment-2218497

**Comment by charlesoestergaard** [comment]
Any updates to this, after the multiplayer has been released?

**Comment by w3b6x9** [comment, marked as answer]
@rumfiske Presence is now available so feel free to upgrade your client to supabase-js v2 rc and check it out. Here's a guide: https://supabase.com/docs/guides/realtime/presence.

**Reply by itsinalis** [reply]
@w3b6x9  I'd love to migrate from v1 to supabase-js v2. 

I tried to, but beyond that I had to change many parts of my app and the logic behind them, I found that the documentation was not enough to help me with the migration (for example Auth: many things have changed but there is no explanation on how to efficiently implement the new logic). 

There are many tutorials on YouTube for building several types of apps with supabase-js v1, which **helped me a lot** when I was building my app in the first place. 

At this point, this is not the case with the new Supabase client. Actually, I found 0 videos on supabase-js v2, and this is the reason I haven't implemented it yet — since I spent many hours trying to figure out what the new possibilities are, what has been deprecated, and what has been replaced, and based on the above, how to **rebuild** my app with the new js client.

If you could create some **practical content** (maybe some 'happy hour' or smth) on how to migrate from v1 to v2, explaining what the new client does, it would help me and others a lot to adopt the v2.

Thank you! 

P.S. The same applies to other newly released features - auth helpers as an example, where I'd love to try them, but I practically can't, since I'll have to spend many hours trying to figure out the setup. I'd love to try them, but I can't afford to lose days testing how it works - and how it should work.

**Reply by charlesoestergaard** [reply]
Thanks, we have merged to this. It works, but still feels like its not 100% correct. Right now we can update a value in db when there is a pressence to make the use "Online" but we cant check if the user has been away for 30minutes. 

If the client closes the website it will destroy the pressence and therefore is offline. When they login, they are online. There is gaps, and sometime online users will show as offline. Can't wait to see a tutorial on this.

**Reply by darkhorsejo** [reply]
That is really a shame because this sort of functionality is common, i was surprise it does come off the shelf
